frontend/containers/SignIn/SignIn.jsx

```diff
@@ -0,0 +1,187 @@
+      */
+     token: proptypes.string
+  }
+
+  constructor(props) {
+    super(props)
+
+    // Is there is a token in the URL.
+    this.validation = new Validation()
+    this.state.isPasswordReset = typeof props.token === 'string' && props.token.length
+    this.state.email = ''
+    this.state.password = ''
+    this.state.passwordConfirm = ''
+    this.state.formDataIsValid = false
+    this.state.error = false
+  }
+
+  componentWillUpdate(nextProps, nextState) {
+    const {actions, state} = this.props
+    const {user, isPasswordReset} = state
+    const nextUser = nextProps.state.user
+    const {resetSuccess} = nextUser
+
+    // If the user is signed in, redirect to the home view.
+    redirectIf(user.remote, '/')
+
+    // Redirect if `resetSuccess` true
+    redirectIf(!user.resetSuccess && (resetSuccess && isPasswordReset), '/sign-in')
+
+    // Reset `isPasswordReset`
+    if (!user.resetSuccess && resetSuccess) {
+      this.setState({isPasswordReset: false})
+    }
+
+    const hasFailed = nextUser.status === Constants.STATUS_FAILED &&
+      nextUser.failedSignInAttempts > 0
+
+    if (nextUser.status === Constants.STATUS_NOT_FOUND) {
+      this.setState({error: 'Authentication API unreachable'})
+    } else if (hasFailed) {
+      this.setState({error:`${isPasswordReset ? 'Passwords don\'t match or invalid token' : 'Email not found or password incorrect'}.`})
+    } else {
+      this.setState({error: false})
+    }
+  }
+
+  render() {
+    const {state, actions, setPagetTitle} = this.props
+    const hasConnectionIssues = state.app.networkStatus !== Constants.NETWORK_OK
+    const {formDataIsValid, error, isPasswordReset} = this.state
+
+    setPagetTitle('Sign In')
+
+    return (
+      <div class={styles.wrapper}>
+        <div class={styles.overlay}>
+          <div class={styles.container}>
+            <form
+              action="/profile"
+              method="POST"
+              onSubmit={this.handleSignIn.bind(this)}
+            >
+              <img class={styles.logo} src="/images/publish.png" />
+
+              {error &&
+                <Banner>{error}</Banner>
+              }
+
+              <div class={styles.inputs}>
+                {!isPasswordReset && (
+                  <div class={styles.input}>
+                    <Label label="Email">
+                      <TextInput
+                        placeholder="Your email address"
+                        validation={this.validation.email}
+                        onChange={this.handleInputChange.bind(this, 'email')}
+                        onKeyUp={this.handleInputChange.bind(this, 'email')}
+                        value={this.state.email}
+                      />
+                    </Label>
+                  </div>
+                )}
+
+                <div class={styles.input}>
+                  <Label label={isPasswordReset ? 'New Password' : 'Password'}>
+                    <TextInput
+                      type="password"
```

Comment:
I'm not sure, I don't think we have any JSX-specific lint rules. I can find out if there's something we can use.